#458174 [C#テキスト]5章演習問題

kurosu-being opened the pull request:
第5章　文字列の操作

1.文字列同士の比較
2.文字列の変換
3文字列の判定
4.文字列の抽出
など、Stringクラスの使用方法を中心に、代表的な文字列操作について学習しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-1-1を実装し、Practice5-2-1のファイルを作成しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-2-1を実装しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-3-1~Practice5-3-3まで実装しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-3-4を実装しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-3-5を仮に実装しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

コメントの修正、変数名の修正を行いました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-4-1において一部コメントの削除を行いました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

コード整形を行いました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

コメントの区切りを追加して修正しました。

ko-kato-being commented on 章末問題5/Practice5-1-1/Program5-1-1.cs:
```diff
@@ -0,0 +1,17 @@
+        static void Main(string[] args) {
+            Console.WriteLine("[○○,■■」という形で2つの文字列を入力してください");
+            string wInput = Console.ReadLine();
+            string wMessage = string.IsNullOrEmpty(wInput) 
+                ? "入力が空です。文字列を入力してください。" : (wInput.Split(',').Length == 2 && string.Compare(wInput.Split(',')[0], wInput.Split(',')[1], true) == 0)
+                ? "一致しています" : "不一致です。\n或いは入力された文字列の数が指定と違います";
```
メッセージの構築を1行で行ってしまうとは、なかなかやりますね 笑
コードを簡潔に記述しようとする意識は素晴らしいです。
ただ、今回は複数行に分けた方が可読性が高まるかなと思いました。

三項演算子を使う時は注意した方が良いです。
個人的には、三項演算子の分岐でさらに三項演算子を使う場合、可読性が落ちるケースが多いと思っています。

kurosu-being replied:
ご指摘ありがとうございます！
複数行に分ける記述に変更いたしました。
三項演算子を適切に使えるよう、三項演算子の分岐でさらに三項演算子を使う場合は、特に可読性を損なっていないかに気を付けたいと思います。

ko-kato-being commented on 章末問題5/Practice5-1-1/Program5-1-1.cs:
```diff
@@ -0,0 +1,17 @@
+        static void Main(string[] args) {
+            Console.WriteLine("[○○,■■」という形で2つの文字列を入力してください");
+            string wInput = Console.ReadLine();
+            string wMessage = string.IsNullOrEmpty(wInput) 
+                ? "入力が空です。文字列を入力してください。" : (wInput.Split(',').Length == 2 && string.Compare(wInput.Split(',')[0], wInput.Split(',')[1], true) == 0)
+                ? "一致しています" : "不一致です。\n或いは入力された文字列の数が指定と違います";
```
「不一致」だったのか「指定数と異なる」だったのか、どちらか判別できた方がユーザーにとって親切かなと思います。

kurosu-being replied:
ご指摘ありがとうございます！
「不一致」だったのか「指定数と異なる」だったのか、ユーザーがどちらか判別できるように変更いたしました。

ko-kato-being commented on 章末問題5/Practice5-1-1/Program5-1-1.cs:
```diff
@@ -0,0 +1,17 @@
+            Console.WriteLine("[○○,■■」という形で2つの文字列を入力してください");
+            string wInput = Console.ReadLine();
+            string wMessage = string.IsNullOrEmpty(wInput) 
+                ? "入力が空です。文字列を入力してください。" : (wInput.Split(',').Length == 2 && string.Compare(wInput.Split(',')[0], wInput.Split(',')[1], true) == 0)
```
`wInput.Split()`  の処理が何度も実行されているのが気になりました。
たぶん無駄なインスタンスを生成しないよう意識してくれたと思います。
その意識は素晴らしいです。
ただ、今回は`wInput.Split()`の戻り値を変数に格納した方が良いかと思います。
その後の処理で必要なもので、無駄ではありませんので。

kurosu-being replied:
ご指摘ありがとうございます！
`string[] wInputArray = wInput.Split(',');`　を追加し、wInput.Split()の処理が何度も出てこないように変更いたしました。
同じ処理を何度も書いている時は特に気を付けたいと思います。

ko-kato-being commented on 章末問題5/Practice5-3-1/Program5-3-1.cs:
```diff
@@ -0,0 +1,14 @@
+namespace Practice5_3_1 {
+    /* "Jackdaws love my big sphinx quartz."という文字列があります。この文字列に対して、
+       以下の問題を解いてください
+       Practice5-3-1 空文字が何文字あるかカウントしてください　*/
```
「空文字」だと、処理と合っていないような気がしますね。
「空白」の数をカウントしたい？

m-hotta-being replied:
それっぽいですね。テキストがおかしい？

kurosu-being replied:
ご指摘ありがとうございます！
問題文を書き違えていました。「空白が何文字あるかカウントしてください」が正しい問題文です。
問題文の書き写し間違いがないかの確認を強化したいと思います。

ko-kato-being replied:
「空白」の数のカウントですね、了解です！
全角の空白「　」はカウントしないですか？
指定した文字が空白かどうかを判定してくれるメソッドがあると思いますので、そちらを使ってみても良いかもしれないですね。

ko-kato-being commented on 章末問題5/Practice5-3-4/Program5-3-4.cs:
```diff
@@ -0,0 +1,17 @@
+        static void Main(string[] args) {
+            string wSentence = "Jackdaws love my big sphinx quartz";
+            string[] wWords = wSentence.Split(' ');
+            var wUnderFourWords = wWords.Where(x => x.Length <= 4);
+            //4文字以下の単語をコンソールに列挙
+            foreach (var wWord in wUnderFourWords) {
```
`wUnderFourWords`の変数は不要かと思います。
foreach のコレクションとして直接指定するのが良いかと。

IEnumerable型の変数は宣言しない方が良いと思います。
遅延評価による副作用もありますので。
https://dev-redmine.being.group/projects/clgaia/wiki/IEnumerableの評価

kurosu-being replied:
ご指摘ありがとうございます！
直接foreach のコレクションとして直接指定する形に変更しました。
また、遅延評価について改めて勉強しなおしました。直接値が必要な場所に書くことで
遅延評価の値が必要な時に呼ばれる、という性質の副作用を回避している、という認識で正しいでしょうか・・・？

ko-kato-being replied:
> また、遅延評価について改めて勉強しなおしました。直接値が必要な場所に書くことで
> 遅延評価の値が必要な時に呼ばれる、という性質の副作用を回避している、という認識で正しいでしょうか・・・？

そうですね、イメージとしては良いと思います。
値が評価されるまでの間に意図しないコードが入り込む余地が無くなる、ということですね。

ko-kato-being commented on 章末問題5/Practice5-3-5/Program5-3-5.cs:
```diff
@@ -0,0 +1,21 @@
+        static void Main(string[] args) {
+            string wSentence = "Jackdaws love my big sphinx quartz";
+            string[] wWords = wSentence.Split(' ');
+            //wStringBuilderObject　←適切な命名が思いつきませんでした
+            var wStringBuilderObject = new StringBuilder();
```
変数名を考える時に、その変数のスコープを意識すると良いです。
スコープとは、その変数を参照できる範囲のことです。
スコープが大きい場合は、いろんなところから参照されることになるので、ある程度具体的な名前が良いです。
逆に、今回のようにスコープが小さい時は、前後の文脈も把握していることを前提にして、簡易的な名前にしてしまっても良いと思います。
私なら、`wBuilder` にしますかね。

m-hotta-being replied:
`wSB` というのも見かけますね。

kurosu-being replied:
@ko-kato-being  @m-hotta-being  
ありがとうございます！
wStringBuilderObject→wBuilderに変更いたしました。
wSBというのも良く使われるのですね。皆さんがどういった命名をされるのか見る機会が少ないので助かります🙇‍♀️

未だに変数の命名で悩みがちですが、スコープを意識したことはなかったです。
全体からみて分かりやすい、という評価だけだと現状どうしても長文の命名になってしまうので
スコープが小さいか大きいかの判断でも命名を行いたいと思います。

ko-kato-being commented on 章末問題5/Practice5-4-1/Program5-4-1.cs:
```diff
@@ -0,0 +1,37 @@
+            Console.WriteLine($"誕生年: {wBorn}年");
+        }
+
+        static string GetValueFromSentence(string vSentence, string vKey) {
+            /*正規表現について理解が不安な部分があるので、
+            自分の考えを書き記します。間違いがある場合はご指摘頂けると幸いです。
+            使用した正規表現
+            (.*?) : '.' が任意の1文字を表し、*? という組み合わせで、文字列が部分一致することを示します。
+            (;|$) ;  ';'または文字列の終わりであることを示します
+
+            この表現によりwMatch に一致する部分を格納し、その値を wMatch.Groups[1].Value で取得しています。*/
+            string wPattern = $"{vKey}=(.*?)(;|$)";
+
+            Match wMatch = Regex.Match(vSentence, wPattern);
+            return wMatch.Success ? wMatch.Groups[1].Value : "見つかりません";
+        }
```
正規表現が正しいか判定できるツールが世の中にはあったりします。
たとえば、↓とか。
https://dev-redmine.being.group/projects/sirius-client/wiki/ツールの便利な使い方#DevToys

不安ならこういうツールを使っても良いかもしれないですね。

ちなみに、私は正規表現を使わずに、Splitを二重に行うかなと思いました。

kurosu-being replied:
ご指摘ありがとうございます！
紹介していただいたDevToysを使って確認すると分かりやすかったです！ありがとうございます。
正規表現以外でも様々な変換やテキスト確認に対応していて面白いですね。

また、章の学習内容的にSplitを使うことを意図していると思うので、Practice5-4-1-SplitにSplitを二重に行う方法で新しく書き直してみました。

ko-kato-being reviewed (changes requested):
ざっとレビューしてみました。
確認お願いします。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-1-1において、三項条件分岐の記述を一部修正し、if条件分岐で修正しました。
Practice5-1-1において、不一致の時と指定数以上の文字列が入力された時にメッセージが異なるよう修正しました。
Practice5-1-1において、wInputArrayを追加しました。

m-hotta-being commented on 章末問題5/Practice5-1-1/Program5-1-1.cs:
```diff
@@ -0,0 +1,24 @@
+    コンソールからの入力の時は、Console.ReadLineメソッドを利用してください。*/
+    class Program {
+        static void Main(string[] args) {
+            Console.WriteLine("[○○,■■」という形で2つの文字列を入力してください");
```
本筋とは関係ないですが……。

これって `[」` であってます？
`「」` とか、 `[]` ではなく？
```
[○○,■■」
```

kurosu-being replied:
ご指摘ありがとうございます！
修正しました。もっと目を凝らして確認します😇

m-hotta-being commented on 章末問題5/Practice5-2-1/Program5-2-1.cs:
```diff
@@ -0,0 +1,14 @@
+        static void Main(string[] args) {
+            Console.WriteLine("整数を入力してください。3桁ごとにカンマ付きの数字文字列に変換します。: ");
+            string wMessage = int.TryParse(Console.ReadLine(), out int wNumber) 
+                ?$"カンマ付きの数字文字列:{wNumber.ToString("#,0")}" : "無効な数字です。";
```
細かいスタイルの話ですが、空白が抜けてます。

```
?$"カンマ付きの数字文字列
```
↓
```
? $"カンマ付きの数字文字列
```

kurosu-being replied:
ご指摘ありがとうございます！
空白を足して修正しました。

m-hotta-being reviewed (changes requested):
ちょっと見てみました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-1-1において、[○○,■■」→「○○,■■」に変更いたしました。
Practice5-2-1において、10行目に空白を追記しました。
Practice5-3-1において、問題文の誤植を修正しました。（空文字→空白）
Practice5-3-4において、11行目のforeach文に直接変数を書き込む形に変更しました。
Practice5-3-5において、wStringBuilder→wBuilderに変更し、コンソールに直接変数を書き込む形に変更しました。
Practice5-3-5-Splitを追加しました。Practice5-3-5の問題をSplitメソッドを二重につかう形で書き直しています。

ko-kato-being commented on 章末問題5/Practice5-1-1/Program5-1-1.cs:
```diff
@@ -0,0 +1,24 @@
+                string wJudge = string.Compare(wInputArray[0].Trim(), wInputArray[1].Trim(), true) == 0
+                      ? "一致しています": "不一致です。";
+                Console.WriteLine(wJudge);
+            }
+            else{
```
少しだけフォーマットがおかしいですね。
コーディング規約的には↓のように書くのが正解です。
`} else {`

kurosu-being replied:
ご指摘ありがとうございます！
正しいフォーマットになるよう修正いたしました。

ko-kato-being replied:
修正ありがとうございました。
`} else{`
惜しいです、非常に細かいのですが、else の後に空白が必要です。
設定ファイルもらうってことなんで、自動整形機能を使っていくと良いですね。

kurosu-being replied:
確認不足でした、失礼いたしました。
再度修正を行いましたので、ご確認のほどよろしくお願いします🙇‍♀️

ko-kato-being commented on 章末問題5/Practice5-4-1-Split/Program5-4-1-Split.cs:
新しいファイルに書いてくれたんですが、Program5-4-1.cs の中で、メソッドを分けるだけでも良いかなと思います。
追加のお題ですが、Novelistクラスを作ってみても良いかもですね。
```
static void Main(string[] args) {
    string wSentence = "Novelist=谷崎潤一郎;BestWork=春琴抄;Born=1886";
    var wNovelist = new Novelist(wSentence);

    Console.WriteLine($"作家  : {wNovelist.○○}");
    Console.WriteLine($"代表作: {wNovelist.○○}");
    Console.WriteLine($"誕生年: {wNovelist.○○}年");
}
```
↑のコードを実行できるようにNovelistクラスを作ってみてください。
wNovelist.○○ はプロパティ名ですので、適切な名前を考えてみてください。

kurosu-being replied:
ご指摘、ご教示いただきありがとうございます！

①「空白」のカウントについて
全角の空白「　」もカウントするように修正いたしました。
②遅延評価について
ご教示ありがとうございます！途中で意図しないコードが割りこむと、こちらの想定していない値に変更されることがあるのですね。実際にコードを書きながら勉強出来ました。ありがとうございます！
③追加のお題
ご指摘ありがとうございます！
復習の際に初めに自分がどう考えたのか、最初の状態を残しておきたかったのでファイルを分けました。見ずらいようでしたら再度修正いたします🙇‍♀️
追加のお題もありがとうございます！
Practice5-4-1-Splitに追加いたしました。（ちょうど次章でDictionaryについてだったので、Key、Valueの扱い方について勉強になりました）

kurosu-being replied:
ご指摘、ご教示いただきありがとうございます！

①「空白」のカウントについて
全角の空白「　」もカウントするように修正いたしました。

②遅延評価について
ご教示ありがとうございます！途中で意図しないコードが割りこむと、こちらの想定していない値に変更されることがあるのですね。実際にコードを書きながら勉強出来ました。

③新しいファイルと追加のお題について
ご指摘ありがとうございます！
復習の際に初めに自分がどう考えたのか、最初の状態を残しておきたかったのでファイルを分けました。見づらいようでしたら再度修正いたします🙇‍♀️

追加のお題もありがとうございます！
Practice5-4-1-Splitに追加いたしました。（ちょうど次章でDictionaryについてだったので、Key、Valueの扱い方について勉強になりました💪）

ko-kato-being replied:
> 復習の際に初めに自分がどう考えたのか、最初の状態を残しておきたかったのでファイルを分けました。見づらいようでしたら再度修正いたします🙇‍♀️

了解です！
黒子さんのリポジトリですので、黒子さんが見やすい書き方にしてもらって構いません:grinning:

ただし、プロジェクトに合流した時は注意が必要です:bulb:
不要なコードは残さないというのが基本です。
ソースコードはGit管理してますので、過去の状態を確認したい場合はコミットをさかのぼります。

kurosu-being replied:
> ただし、プロジェクトに合流した時は注意が必要です💡
> 不要なコードは残さないというのが基本です。
> ソースコードはGit管理してますので、過去の状態を確認したい場合はコミットをさかのぼります。

現状の私のコミットするタイミングが「キリが良い時にコミットする」という区切りであった為、
Git管理をしている利点を考えておりませんでした。
「状態を保存する」ことを考えれば、後に必要になる状態があればその時点でコミットを行い、
後に遡りやすいコメントを残すのが適切ですね。
プロジェクトに合流した際に不要なコードを残す癖が付かないよう、コメントの書き方やGitの扱いにも慣れていきたいと思います🙋‍♀️

m-hotta-being commented on 章末問題5/Practice5-3-5/Program5-3-5.cs:
【ただのコメント、修正不要です】
* 読み取り箇所が 1 箇所しかない変数を作らない。

なんてことを試しにやってみると、また違った書き方の模索ができると思います。
やり過ぎると読みにくくなりますが、そういった書き方がされたコードも読めるように、練習してみると良いと思います。

ko-kato-being replied:
便乗して。
無駄な変数を作らないことによって、変数名を考えなくて良いというメリットもありますねー。

kurosu-being replied:
ご教示ありがとうございます！
Wiki の 4. Visual Studio の設定については、後日宮原さんからインポートするファイルを配布して頂く予定です。
どういったフォーマットなのか把握しつつ、ツールを使用していこうと思います。

> - 読み取り箇所が 1 箇所しかない変数を作らない。
> 
> なんてことを試しにやってみると、また違った書き方の模索ができると思います。

ありがとうございます！省略すべき変数かどうかの判断において検討していきます。

m-hotta-being reviewed (approved):
修正ありがとうございます。
良いと思います。

[Wiki の 4. Visual Studio の設定を統一する](https://dev-redmine.being.group/projects/sirius-client/wiki/%E9%96%8B%E7%99%BA%E7%92%B0%E5%A2%83%E3%81%AE%E6%A7%8B%E7%AF%89%E6%89%8B%E9%A0%86#4-Visual-Studio-%E3%81%AE%E8%A8%AD%E5%AE%9A%E3%82%92%E7%B5%B1%E4%B8%80%E3%81%99%E3%82%8B) とかって研修ではどうしてるんでしょう？まだ入れてないでしょうか？
フォーマットはツールに任せてしまったほうが、本当に重要なところに集中できてよいと思います。

もし何のことか分からなければ 宮原さん とか近くの先輩に聞いてみてください。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-1-1において、フォーマットのズレを修正しました。
Practice5-3-1において、全角やタブの空白もカウントできるよう修正しました。
Practice5-4-1-Splitにおいて追加のお題を実装しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-3-1において、追加したメソッドにコメントを追加しました。
Practice5-4-1で-Splitにおいて、追加したメソッドにコメントを追加しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Novelistクラスのプロパティ、コンストラクタにコメントを追加しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-4-1-Splitにおいて、メソッドのコメントを追加しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-3-1において、不要な空行を削除しました。

ko-kato-being commented on 章末問題5/Practice5-3-1/Program5-3-1.cs:
```diff
@@ -0,0 +1,30 @@
+
+            Console.WriteLine($"空白文字の個数: {wCount}");
+        }
+        /// <summary>
+        /// 空白をカウントするメソッド
+        /// </summary>
+        /// <param name="vInput">受けとる文字列</param>
+        /// <returns>空白の個数</returns>
+        static int CountSpaces(string vInput) {
+            int wCount = 0;
+            foreach (char wChar in vInput) {
+                if (char.IsWhiteSpace(wChar)) {
+                    wCount++;
+                }
+            }
+            return wCount;
+        }
```
char.IsWhiteSpace() を使うことによって、様々な「空白」に対応することができましたね、素晴らしい！
空白とはそもそも何なのか、調べていくと「文字コード」というワードが出てくると思います。
開発してるとたまにこの言葉に遭遇すると思いますので、時間のある時にどういうものなのか調べてみると良いかもしれません。
(すでにバッチリですかね？😇)

ちなみにですが、既存のCount()メソッドを使わずに、新たにメソッドを作ったのは理由があったりしますか？

kurosu-being replied:
文字コードについてはIsWhiteSpaceメソッドを選択する際に調べましたが、Unicodeが最も網羅的に文字が扱えるからこれで行こうと考えた時点で検索をやめていました。改めて調べると他の符号化文字集合やUTF-8について知らないことが多そうです。これから学習を進めていこうと思います！

既存のCount()メソッドを使わなかった理由ですが、結論から申し上げますとリファクタリング不足です😇
（全角やTabの空白も削除するメソッドを用いた時、なんらかの条件分岐が発生するのではないかと考え、
　最初にメソッドの枠だけ用意してから書き始めましたが、必要ありませんでした。）
よって、Count()メソッドを用いて簡潔に書くよう修正を行いました。
お手数をおかけしますが、再度ご確認いただけると幸いです。

ko-kato-being replied:
修正ありがとうございます、OKです！
どういう経緯で作ったメソッドだったのか、とてもよくわかりました。
説明スキル高いねー:innocent:

ko-kato-being commented on 章末問題5/Practice5-4-1-Split/Novelist.cs:
```diff
@@ -0,0 +1,33 @@
+        /// </summary>
+        /// <param name="vSentence">受け取る文字列</param>
+        public Novelist(string vSentence) {
+            var parts = vSentence.Split(';')
+                .Select(x => x.Split('='))
+                .ToDictionary(p => p[0], p => p[1]);
+            if (parts.ContainsKey("Novelist"))
+                this.Writer = parts["Novelist"];
+            if (parts.ContainsKey("BestWork"))
+                this.BestWork = parts["BestWork"];
+            if (parts.ContainsKey("Born") && int.TryParse(parts["Born"], out int wBornYear))
+                this.Born = wBornYear;
```
Born は int 型のプロパティにしたんですね。
とても良いと思います！
プログラミング始めて間もないと、string型で受けとったからstring型の変数に格納するって、やっちゃいがちな気がしますが、黒子さんは筋が良いですね😊

parts がコーディング規約違反なのが惜しいです！

kurosu-being replied:
ありがとうございます！💪
parts→wPartsに修正いたしました。

ko-kato-being commented on 章末問題5/Practice5-4-1-Split/Novelist.cs:
```diff
@@ -0,0 +1,33 @@
+﻿using System.Linq;
+
+namespace Practice5_4_1_Split {
+    class Novelist {
+        /// <summary>
+        /// 作家
+        /// </summary>
+        public string Writer { get; private set; }
+        /// <summary>
+        /// 代表作
+        /// </summary>
+        public string BestWork { get; private set; }
+        /// <summary>
+        /// 誕生年
+        /// </summary>
+        public int Born { get; private set; }
```
クラスにもコメントを書きましょう。

プロパティ名がちょっと微妙な気がしました。
「Writer」で表現したいのは作家の名前かなと思います。
私なら「Name」というプロパティ名にしそうです。
参考までに～。

あ、あと、コンストラクタでしか値を設定しないプロパティはセッターを省略できます。
こちらも参考までに！

m-hotta-being replied:
> あ、あと、コンストラクタでしか値を設定しないプロパティはセッターを省略できます。

補足ですー。
`private set;` の有無で、たぶん思っているより挙動が大きく変わります。
試しに下記のメソッドを追加して、セッターを省略したり元に戻したりしてみてください。
セッターをなぜ省略するのかが少しわかるかもしれません。

```
public void SetWriter(string vName) {
    Writer = vName;
}
```

kurosu-being replied:
@ko-kato-being 
ご質問ありがとうございます！
①クラスにコメントを追加
②プロパティ名をWriter→Nameに変更
③セッターの省略
を行いました。

> private set; の有無で、たぶん思っているより挙動が大きく変わります。
> 試しに下記のメソッドを追加して、セッターを省略したり元に戻したりしてみてください。

ありがとうございます！試してみると、セッターが無い状態であるとエラーメッセージでも「読み取り専用の為変更できない」という旨をしめしてくれ分かりやすかったです。
セッターをprivateにするかどうか、またはセッターが不要なのか判断する為に、メソッドで何か値を書き換えようとしているか、コンストラクタでしか値を設定しないプロパティかどうかの確認を行いたいと思います。

ko-kato-being replied:
セッター非公開 < private < public の順にスコープは大きくなります。
基本的には最小のスコープにしておくのがおすすめです。
必要になった時にスコープを広げるのが良いと思います。
これは「YAGNI(やぐに)」という原則で、私は好きな考え方です:blush:

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-1-1において、elseの右の空白を削除しました。
Practice5-3-1において、CountSpacesメソッドを削除し、LINQを用いて書き直しました。
Practice5-4-1において、GetValueFromSentenceメソッドのコメントを追加しました。
Practice5-4-1-Spliにおいて、Novelistクラスのコメントを追加しました。
Practice5-4-1-Splitにおいて、プロパティ名をWriter→Nameに変更しました。
Practice5-4-1-Splitにおいて、GetPartsValueのコメントを修正しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-4-1において、正規表現についてのメモ書きコメントを削除しました。

kurosu-being committed:
#458174 [C#テキスト]5章演習問題

Practice5-4-1-Splitにおいて、Novelistクラスのセッターを削除しました。
